pipenv/project.py

```diff
@@ -685,8 +685,8 @@ def create_pipfile(self, python=None):
         from .vendor.pip_shims.shims import (
             ConfigOptionParser, make_option_group, index_group
         )
-
-        config_parser = ConfigOptionParser(name=self.name)
+        # Inherit the pip's index configuration of install command.
+        config_parser = ConfigOptionParser(name="install")
         config_parser.add_option_group(make_option_group(index_group, config_parser))
         install = config_parser.option_groups[0]
```

Comment:
```suggestion
        install = config_parser.option_groups[0]
        indexes = command.cmd_opts.get_option("--extra-index-url").default
```

and drop the `indexes` defined in the line below